### PR TITLE
Prepare for localization automation

### PR DESCRIPTION
**Pull request by dougbu**
First, we need to keep the `id`'s up to date&hellip;

* Cleanup SPA templates slightly

  - add missing `description`s
  - add `id` for `postActions`
  - add and correct `$schema` properties
  - generalize vs-2017.3.host.json filename to ide.host.json

* Add templatestrings.*.json files
  - step 3/4 (yeah, not just of 3) for https://github.com/dotnet/aspnetcore/issues/36998 in this repo
  - without this, id's are **not** automatically kept up to date
  - see <https://github.com/dotnet/templating/wiki/Localization>
  - see also dotnet/aspnetcore#38867 (a similar update in dotnet/aspnetcore)
  
* Move existing translations from old files to new
  - {lang}/strings.json to templatestrings.{lang}.json
  
* Remove redundant strings.json files
  - templatestrings.{lang}.json files are enough
  - see also dotnet/aspnetcore#40128 (a similar update in dotnet/aspnetcore)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,8 +1,15 @@
 # How to contribute
 
+## Update existing templates
+
+You can submit changes for templates in this repo by submitting a pull request. If you make changes to any
+`content/*/.template.config/template.json` files, build locally and include any
+`content/*/.template.config/localize/` changes in your pull request. (Your build may update the strings in those
+files for later localization.)
```
Comment (dougbu) [inline CONTRIBUTING.md:8]: This is about the same text as in dotnet/aspnetcore#42587

File: src/content/Angular-CSharp/.template.config/ide.host.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "$schema": "http://json.schemastore.org/vs-2017.3.host",
+  "$schema": "http://json.schemastore.org/ide.host",
```
Comment (dougbu) [inline src/content/Angular-CSharp/.template.config/ide.host.json:2]: @phenning was I correct to rename this file and src/content/React-CSharp/.template.config/vs-2017.3.host.json❔

File: src/Microsoft.DotNet.Web.Spa.ProjectTemplates.csproj [hunk 3]
```diff
@@ -22,7 +22,13 @@
 
   <Choose>
```
Comment (dougbu) [inline src/Microsoft.DotNet.Web.Spa.ProjectTemplates.csproj:23]: Confirmed the dev build cases here work in local testing within the spa-templates submodule in aspnetcore as well as in an "isolated" spa-templates clone.

**Review by bekir-ozturk — COMMENTED** [review]
You probably want to add `displayName` property to all the parameter-type symbols as well. Otherwise, the symbol `ExcludeLaunchSettings` will show up in VS as `ExcludeLaunchSettings` instead of `Exclude launch settings`.

Here is an example from templating repo: https://github.com/dotnet/templating/blob/main/template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ConsoleApplication-CSharp/.template.config/template.json#L22-L28

**Comment by dougbu** [comment]
> You probably want to add displayName property to all the parameter-type symbols as well.

@mkArtakMSFT @adityamandaleeka @rafikiassumani-msft, I _could_ do this but it would be pretty blind. Should we wait i.e. defer to follow-ups from your teams, split between teams now, or go w/ my blunders for now❔

File: src/content/Angular-CSharp/.template.config/ide.host.json (hunk 2 above)
Comment (dougbu) [reply src/content/Angular-CSharp/.template.config/ide.host.json:2]: /ping @phenning

**Comment by mkArtakMSFT** [comment]
Thanks for handling this, @dougbu!
I'm wondering if a faster way would be to merge this and get CTI to report all the issues where the localization didn't work?

**Comment by mkArtakMSFT** [comment]
@MackinnonBuck what do you think will take to get this change validated? I mean, how much time will it require?

**Comment by dougbu** [comment]
In terms of speed, adding `.../displayName` properties might not take that much time. We have less than 50 strings in src/content/Angular-CSharp/.template.config/localize/templatestrings.en.json and src/content/React-CSharp/.template.config/localize/templatestrings.en.json combined and a lot of the symbols look like duplicates.

Having CTI complain about "`ExcludeLaunchSettings` instead of `Exclude launch settings`" also doesn't seem viable since this PR shouldn't change the lack of display names in the template.json files. Why haven't they complained about what's displayed in VS before❔

**Comment by mkArtakMSFT** [comment]
They did and you see several such issues linked from the issue tracking the localization work.

**Comment by dougbu** [comment]
I see the issues linked to #36998 but nothing sounds similar to the "`ExcludeLaunchSettings` instead of `Exclude launch settings`" case @bekir-ozturk described. Is that because CTI hasn't been looking at the two SPA templates❔

**Comment by mkArtakMSFT** [comment]
Not sure where that option is exposed, so they may have not seen it either or may have simply missed.

File: src/content/Angular-CSharp/.template.config/ide.host.json (hunk 2 above)
Comment (phenning) [reply src/content/Angular-CSharp/.template.config/ide.host.json:2]: Additional cleanup: the following can be removed from the ide.host.json file:

1. name and description properties (these will be pulled in from templatestrings.json.
2. learn more link can be removed, it is no longer used
3. uiFilters section can be removed - no longer uesed
4. supportedAuthentications can be removed - no longer used.. we just use the "auth" in the template.json
5. ports - no longer used - we use the port generators in template.json
6. excludeLaunchSetttings - no longer used
7. minFullFrameworkVersion - no longer used.

Note that the same cleanup has been done to the non-spa templates, but it looks like @DamianEdwards  and I forgot about the spa templates.

**Comment by dougbu** [comment]
> You probably want to add `displayName` property to all the parameter-type symbols as well. Otherwise, the symbol `ExcludeLaunchSettings` will show up in VS as `ExcludeLaunchSettings` instead of `Exclude launch settings`.

I see minimal use of `displayName` in the dotnet/aspnetcore repo. Is it true that the property matters only for parameters made visible by ide.host.json❔ Just want to make sure I catch all the places where we need display names.

**Comment by dougbu** [comment]
- [ ] name and description properties (these will be pulled in from templatestrings.json.
- [ ] learn more link can be removed, it is no longer used
- [ ] uiFilters section can be removed - no longer uesed
- [ ] supportedAuthentications can be removed - no longer used.. we just use the "auth" in the template.json
- [ ] ports - no longer used - we use the port generators in template.json
- [ ] excludeLaunchSetttings - no longer used
- [ ] minFullFrameworkVersion - no longer used.

**Comment by dougbu** [comment]
If I'm understanding correctly, the ide.host.json files are now short enough we don't need more `displayName` settings in template.json files.

----

In the dotnetcli.host.json files, can `"shortName": ""` be removed❔ I'm guessing that means the same thing as leaving the property out.

**Review by bekir-ozturk — APPROVED** [review]
LGTM

**Comment by dougbu** [comment]
I'm not seeing what you suggest @phenning. It instead appears like `longName` is what matters for `dotnet new` e.g.

``` text
> dotnet new angular -h
...
Template options:
  -au, --auth <Individual|None>    The type of authentication to use
                                   Type: choice
                                     None        No authentication
                                     Individual  Individual authentication
                                   Default: None
  --exclude-launch-settings        Whether to exclude launchSettings.json in the generated template.
                                   Type: bool
                                   Default: false
  --no-restore                     If specified, skips the automatic restore of the project on create.
                                   Type: bool
                                   Default: false
  -P, --ProxyPort <ProxyPort>      Port number to use for the SPA front-end development proxy.
                                   Type: integer
  --no-https                       Whether to turn off HTTPS. This option only applies if Individual, IndividualB2C,
                                   SingleOrg, or MultiOrg aren't used for --auth.
                                   Type: bool
                                   Default: false
  -uld, --use-local-db             Whether to use LocalDB instead of SQLite. This option only applies if --auth
                                   Individual or --auth IndividualB2C is specified.
                                   Type: bool
                                   Default: false
  -f, --framework <net5.0|net7.0>  The target framework for the project.
                                   Type: choice
                                     net7.0  Target net7.0
                                     net5.0  Target net5.0
                                   Default: net7.0
  --use-program-main               Whether to generate an explicit Program class and Main method instead of top-level
                                   statements.
                                   Type: bool
                                   Default: false
```

So, the worst bug here is w/ `--ProxyPort` versus `--proxy-port` and I'm not going to worry about it because the short and long names in the CLI aren't localized. That's something others on @mkArtakMSFT's team can handle, once they check the CLI and VS in more detail.

**Comment by dougbu** [comment]
Oddly @bekir-ozturk's approval isn't enough. Could someone else please approve❔

**Review by mkArtakMSFT — APPROVED** [review]

**Pull request merged**